warehouse/macaroons/caveats/__init__.py

```diff
@@ -104,7 +104,7 @@ class RequestUser(Caveat):
     user_id: StrictStr
 
     def verify(self, request: Request, context: Any, permission: str) -> Result:
-        if not isinstance(request.identity, UserTokenContext):
+        if not isinstance(request.identity, UserContext):
             return Failure("token with user restriction without a user")
 
```

Comment:
Should there be a new extra check here?:
```python
        if request.identity.macaroon is None:
            return Failure("token with user restriction without a macaroon")
```
or will that never happen?

Reply:
I _think_ this can't happen, since the only place we end up checking the caveats is in `MacaroonSecurityPolicy.permits`, where `request.identity` will invariably either be an `PublisherTokenContext` or a `UserContext` that always has its macaroon set.

That being said, I see no harm in it as a sanity check either 🙂.

Reply:
added!